lawnchair/src/app/lawnchair/util/ClipboardUtils.kt

```diff
@@ -0,0 +1,45 @@
+package app.lawnchair.util
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Context
+import android.widget.Toast
+import com.android.launcher3.R
+
+fun copyToClipboard(
+    context: Context,
+    text: String,
+    label: String = text,
+    toastMessage: String? = context.getString(R.string.copied_toast),
+) {
+    val clipboardManager: ClipboardManager = context.requireSystemService()
+    val clip = ClipData.newPlainText(label, text)
+    clipboardManager.setPrimaryClip(clip)
+    toastMessage?.let {
+        Toast.makeText(
+            context,
+            it,
+            Toast.LENGTH_SHORT,
+        ).show()
+    }
+}
+
+fun String.copyToClipboard(
+    context: Context,
+    label: String = this,
+    toastMessage: String? = context.getString(R.string.copied_toast),
+) {
+    copyToClipboard(
+        context = context,
+        text = this,
+        label = label,
+        toastMessage = toastMessage,
+    )
+}
+
+fun getClipboardContent(
+    context: Context,
+): String? {
+    val clipboardManager: ClipboardManager = context.requireSystemService()
+    return clipboardManager.primaryClip?.getItemAt(0)?.text?.toString()
+}
```

Comment:
Make `getClipboardContent` as a extension for `Context`?

```suggestion
fun Context.getClipboardContent(): String? =
    requireSystemService<ClipboardManager>().primaryClip?.getItemAt(0)?.text?.toString()
```

Reply:
I can make an extra extension function but I think the main function should not be one.

Reply:
One should be enough, could be invoked by Kotlin and Java.

Reply:
Ill just keep the base non-extension function, I usually avoid making extension functions for too generic classes eg context/string unless they need to be used many times.